libraries/rush-lib/src/logic/operations/OperationMetadataManager.ts

```diff
@@ -45,19 +47,24 @@ export class OperationMetadataManager {
   private _relativeErrorLogPath: string;
 
   public constructor(options: IOperationMetadataManagerOptions) {
-    const { rushProject, phase } = options;
+    const { rushProject, phase, operation } = options;
     const { projectFolder } = rushProject;
 
     const identifier: string = phase.logFilenameIdentifier;
-    this._metadataFolder = `${RushConstants.projectRushFolderName}/${RushConstants.rushTempFolderName}/operation/${identifier}`;
+
+    const operationPath: string | undefined = operation.shard
+      ? `${operation.name}_${operation.shard.current}`
```

Comment:
The shard should be part of the operation name, which is in turn just the forwarded runner name.

Reply:
Updated in [7d34e74](https://github.com/microsoft/rushstack/pull/4652/commits/7d34e74bc94de34fb1bafe0c89cde48b7d01c7de)

Reply:
Follow up question here, this currently only depends on `phase.logFilenameIdentifier` for the directory name. Should I move that into Operation instead or just use the operation name here? If we don't use the operation name, shards aren't differentiated here - but this might be a breaking change for projects that upgrade and expect the old path to restore correctly.

Reply:
Adjusted to a new `operationMetadataManager.logFilenameIdentifier` which is used for all cacheable operation logs